system/include/emscripten/threading.h

```diff
@@ -332,10 +332,17 @@ EMSCRIPTEN_RESULT emscripten_wait_for_call_i(em_queued_call *call, double timeou
 
 void emscripten_async_waitable_close(em_queued_call *call);
 
-// Queues the given function call to be performed on the specified thread.
-void emscripten_async_queue_on_thread_(pthread_t target_thread, EM_FUNC_SIGNATURE sig, void *func_ptr, void *satellite, ...);
+void _emscripten_call_on_thread(int force_async, pthread_t target_thread, EM_FUNC_SIGNATURE sig, void *func_ptr, void *satellite, ...); // internal
 
-#define emscripten_async_queue_on_thread(target_thread, sig, func_ptr, satellite, ...) emscripten_async_queue_on_thread_((target_thread), (sig), (void*)(func_ptr), (satellite),##__VA_ARGS__)
+// Queues the given function call to be performed on the specified thread. If
+// it is already on the right thread, it will just execute it; otherwise it
```

Comment:
"If it is already on the right thread" reads a bit vague. Perhaps instead `"If the target thread is the same as the calling thread, the function will be immediately executed before returning."?

Reply:
Thanks, yeah, that was unclear... fixing.